### PR TITLE
[charts/sn-platform] Add prometheus pod name to sts env

### PR DESCRIPTION
**Pull request by yuweisung**
### Motivation

Prometheus HA uses $POD_NAME to detect which pod is active. Add env POD_NAME to prometheus statefulset template.

### Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

### Documentation

Check the box below.

Need to update docs? 

- [ ] `doc-required` 
  
  (If you need help on updating docs, create a doc issue)
  
- [x] `no-need-doc` 
  
  (Please explain why)
  
- [ ] `doc` 
  
  (If this PR contains doc changes)

### CODE REVIEW

File: charts/sn-platform/values.yaml [hunk 1]
```diff
@@ -1789,7 +1793,7 @@ prometheus:
       periodSeconds: 10
   customRelabelConfigs: []
   extraScrapeConfigs: []
-  extraCustomArgs: {}
+  extraCustomArgs:
```
Comment (ericsyh) [inline charts/sn-platform/values.yaml:1796]: Why delete `{}` here?

**Review by ericsyh — COMMENTED** [review]
left comments

File: charts/sn-platform/values.yaml [hunk 2]
```diff
@@ -1771,6 +1771,10 @@ prometheus:
     # https://prometheus.io/docs/prometheus/latest/storage/#operational-aspects
     - --storage.tsdb.wal-compression
   scrapeInterval: 15s
+  externalLabels:
```
Comment (ericsyh) [inline charts/sn-platform/values.yaml:1774]: @yuweisung These are testing values? Should to remove.

**Review by ericsyh — COMMENTED** [review]
Left comments, should remove testing values.

**Review by yuweisung — COMMENTED** [review]
remove test value and add empty map

**Review by yuweisung — COMMENTED** [review]
remove test

**Review by zerbitx — APPROVED** [review]

**Review by ericsyh — APPROVED** [review]

**Pull request merged**